### PR TITLE
Complete dice work

### PR DESCRIPTION
**Pull request by grimreaper**
This finishes the work to modularize and abstract out the dice work. It also kills DiceEx which isn't used and should be rebuilt as a DiceConfiguration.

There is more than can be done: making toFormula be generated from the collector + modifiers. The Dice, SystemDie, and related classes could be pulled apart and moved into the user classes, but this is good enough for now.

### CODE REVIEW

**Comment by cpmeister** [comment]
This has  conflicts

**Comment by cpmeister** [comment]
The new files need the license headers

File: code/src/java/gmgen/plugin/dice/Die.java [hunk 1]
```diff
@@ -20,45 +20,34 @@
...
  *  Created on January 24, 2002, 11:15 AM
  */
-package gmgen.plugin;
+package gmgen.plugin.dice;
 
 import java.util.Random;
 
-/** Abstract class describing a die of any kind
- * @author Soulcatcher
- * @since May 24, 2003
```
Comment (cpmeister) [inline code/src/java/gmgen/plugin/dice/Die.java]: Why are you removing the author tag?

Comment (grimreaper) [inline code/src/java/gmgen/plugin/dice/Die.java]: This particular file has been basically stripped to the bare minimum and no longer directly reflects the work of the author. I can restore of required, but its of questionable value (vcs stores this history)

**Comment by cpmeister** [comment]
Add license headers to the new files and resolve the conflicts and I'll merge it.

**Comment by grimreaper** [comment]
license headers added

**Pull request merged**